telegram/ext/_chatjoinrequesthandler.py

```diff
@@ -43,17 +44,41 @@ async def callback(update: Update, context: CallbackContext)
 
             The return value of the callback is usually ignored except for the special case of
             :class:`telegram.ext.ConversationHandler`.
+        chat_id(:obj:`int` | Collection[:obj:`int`], optional): Filters requests to allow only
+            those which are from a specified chat ID.
+
+            .. versionadded:: 20.0
         block (:obj:`bool`, optional): Determines whether the return value of the callback should
             be awaited before processing the next handler in
             :meth:`telegram.ext.Application.process_update`. Defaults to :obj:`True`.
 
     Attributes:
         callback (:term:`coroutine function`): The callback function for this handler.
+        chat_ids(:frozenset(:obj:`str`), optional):
+            Which chat ID(s) to allow through.
+
+            .. versionadded:: 20.0
         block (:obj:`bool`): Determines whether the callback will run in a blocking way..
 
     """
 
-    __slots__ = ()
+    __slots__ = ("chat_ids",)
+
+    def __init__(
+        self,
+        callback: HandlerCallback[Update, CCT, RT],
+        chat_id: SCT[int] = None,
+        block: DVInput[bool] = DEFAULT_TRUE,
+    ):
+        super().__init__(callback, block=block)
+
+        if chat_id is not None:
+            if isinstance(chat_id, int):
+                chat_id = frozenset({chat_id})
+            else:
+                chat_id = frozenset(chat_id)
+
+        self.chat_ids = chat_id
```

Comment:
This should be a private attribute. You probably want to implement it with a setter, so you can check if either chat ids or usernames are already set, see https://github.com/python-telegram-bot/python-telegram-bot/blob/master/telegram/ext/filters.py#L627-L641